dvc/tree/dvc.py

```diff
@@ -149,17 +133,14 @@ def _fetch_dir(
         # pull dir cache if needed
         out.get_dir_cache(**kwargs)
 
-        # pull dir contents if needed
-        if self.fetch and out.changed_cache(filter_info=filter_info):
-            used_cache = out.get_used_cache(filter_info=filter_info)
-            downloaded = self.repo.cloud.pull(used_cache, **kwargs)
-            if download_callback:
-                download_callback(downloaded)
+        dir_cache = out.dir_cache
+        hash_info = out.cache.save_dir_info(dir_cache)
+        if hash_info != out.hash_info:
+            from dvc.output.base import CollectCacheError
```

Comment:
A temporary workaround, we should be able to open cache dir remotely without fetching it.